frontend/src/components/widgets/CameraInput/CameraInput.tsx

```diff
@@ -0,0 +1,579 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import axios from "axios"
+import _ from "lodash"
+
+import { X } from "@emotion-icons/open-iconic"
+import Icon from "src/components/shared/Icon"
+import {
+  FileUploaderState as FileUploaderStateProto,
+  UploadedFileInfo as UploadedFileInfoProto,
+  CameraInput as CameraInputProto,
+} from "src/autogen/proto"
+
+import { FormClearHelper } from "src/components/widgets/Form"
+import { FileUploadClient } from "src/lib/FileUploadClient"
+import { WidgetStateManager } from "src/lib/WidgetStateManager"
+import { logError } from "src/lib/log"
+import {
+  WidgetLabel,
+  StyledWidgetLabelHelp,
+} from "src/components/widgets/BaseWidget"
+import TooltipIcon from "src/components/shared/TooltipIcon"
+import { Placement } from "src/components/shared/Tooltip"
+
+import {
+  UploadFileInfo,
+  UploadedStatus,
+  UploadingStatus,
+} from "../FileUploader/UploadFileInfo"
+
+import CameraInputButton from "./CameraInputButton"
+import WebcamComponent from "./WebcamComponent"
+
+import { StyledBox, StyledCameraInput, StyledSpan } from "./styled-components"
+
+export interface Props {
+  element: CameraInputProto
+  widgetMgr: WidgetStateManager
+  uploadClient: FileUploadClient
+  disabled: boolean
+  width: number
+}
+
+type FileUploaderStatus =
+  | "ready" // FileUploader can upload or delete files
+  | "updating" // at least one file is being uploaded or deleted
```

Comment:
nit: some of these explanatory comments should probably be slightly rewritten for this widget